libs/application/template-api-modules/src/lib/modules/templates/transport-authority/order-vehicle-license-plate/order-vehicle-license-plate.service.ts

```diff
@@ -91,51 +86,132 @@ export class OrderVehicleLicensePlateService extends BaseTemplateApiService {
       )
     }
 
+    // Case: count > 20
+    // Display search box, validate vehicle when permno is entered
+    if (totalRecords > 20) {
+      return {
+        totalRecords: totalRecords,
+        vehicles: [],
+      }
+    }
+
+    const resultData = result.data || []
+
+    const vehicles = await Promise.all(
+      resultData.map(async (vehicle) => {
+        // Case: 20 >= count > 5
+        // Display dropdown, validate vehicle when selected in dropdown
+        if (totalRecords > 5) {
+          return this.mapVehicle(auth, vehicle, false)
+        }
+
+        // Case: count <= 5
+        // Display radio buttons, validate all vehicles now
+        return this.mapVehicle(auth, vehicle, true)
+      }),
+    )
+
+    return {
+      totalRecords: totalRecords,
+      vehicles: vehicles,
+    }
+  }
```

Comment:
Setja á long tail að búa til util fall getVehiclesForApplication sem tekur in mapVehicle fall, sækir vehicles og skilar mappedVehicles eftir því hvað notandinn á mörg ökutæki?

Finnst við vera að gera basically sama hlutinn í svo mörgum umsóknum